staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/controller.go

```diff
@@ -400,10 +339,13 @@ func (c *celAdmissionController) Validate(
 }
 
 func (c *celAdmissionController) HasSynced() bool {
-	return c.policyBindingController.HasSynced() &&
-		c.policyDefinitionsController.HasSynced()
+	return c.policyController.HasSynced() && c.definitions.Load() != nil
 }
 
 func (c *celAdmissionController) ValidateInitialization() error {
-	return c.validatorCompiler.ValidateInitialization()
+	return c.policyController.ValidateInitialization()
+}
+
+func (c *celAdmissionController) refreshPolicies() {
```

Comment:
Is `refreshPolicies()` called outside of the initial sync? I did a string search for calls and couldn't find anything obvious?

If not, what triggers the policies to refresh on controller events?

Reply:
https://github.com/kubernetes/kubernetes/blob/3ac834910adb96411866711780ffba50922e532f/staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/controller.go#L138-L148

The `wait.Until` will keep calling `c.refreshPolicies()` every second until the context is cancelled. I went with polling since I didn't think taking the lock once per second is a large cost for the simplicity. Open to alternative ideas

Reply:
Note that since https://github.com/kubernetes/kubernetes/pull/113985, the `HasSynced` function also will wait for the initial listing to have been reconciled.

Reply:
Gotcha, thanks!